### ISSUE TITLE
Topshelf halts service recovery after third attempt

### ISSUE THREAD

**Issue by mikkelxyz** [Topshelf/Topshelf#404]
I am using Topshelf to create a Windows Service. This service will attempt recovery the first 3 failures, but after that, it no longer works.

Inspecting the service in Services on the host reveals:

    First Failure:          Restart the Service
    Second Failure:         Restart the Service
    Subsequent Failures:    Restart the Service
    Reset fail count after: 1 days
    Restart service after:  2 minutes

The service recovery code looks like this:

    f.EnableServiceRecovery(r =>
    {
        r.RestartService(2);
        r.RestartService(5);
        r.RestartService(5);
        r.OnCrashOnly();
        r.SetResetPeriod(1);
    });

Inspecting the Event Log shows the following messages after failed recovery:

    The MyService service terminated unexpectedly.  It has done this 1 time(s).  The following corrective action will be taken in 120000 milliseconds: Restart the service.
    The MyService service terminated unexpectedly.  It has done this 2 time(s).  The following corrective action will be taken in 300000 milliseconds: Restart the service.
    The MyService service terminated unexpectedly.  It has done this 3 time(s).  The following corrective action will be taken in 300000 milliseconds: Restart the service.
    The MyService service terminated unexpectedly.  It has done this 4 time(s).

As is evident from the above, the fourth time does not trigger recovery.

Is this a Windows error, a Topshelf issue, or is there something wrong in my configuration?

**Comment by mikkelxyz** [comment]
I have tried to add an extra r.RestartService and this changes to behaviour so that the service is restarted 4 times and stops on the 5th attempt. I do not think this is agreement with the documentation where it says that "you can have up to three of these" and "the last one will act for all subsequent failures".

**Comment by LarsKemmann** [comment]
I ran into this same issue today (.NET Framework 4.6.2, Topshelf 4.0.3). Even after I waited for over 24 hours after the fourth unexpected termination (per the `r.SetResetPeriod(1)` setting), Windows took no action, no further relevant events were logged in the Event Log, and the service remained unrecovered.

This is actually a pretty serious problem. Windows does have the option (based on looking at the Services control panel applet) to set a continuous service restart policy, say every 5 minutes. Does that option actually work, and if so, can Topshelf be updated to handle setting that policy correctly?
